File: app/src/main/java/com/awscherb/cardkeeper/data/service/ScannedCodeService.java
package com.awscherb.cardkeeper.data.service;


import com.awscherb.cardkeeper.data.model.ScannedCode;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Observable;
import io.reactivex.Single;


public interface ScannedCodeService {

    Single<ScannedCode> getScannedCode(long codeId);
    Observable<List<ScannedCode>> listAllScannedCodes();

    Single<ScannedCode> addScannedCode(ScannedCode scannedCode);

    Single<ScannedCode> updateScannedCode(ScannedCode scannedCode);

    Completable deleteScannedCode(ScannedCode scannedCode);

}

File: app/src/main/java/com/awscherb/cardkeeper/data/service/ScannedCodeService.kt
package com.awscherb.cardkeeper.data.service

import com.awscherb.cardkeeper.data.entity.ScannedCodeEntity
import com.awscherb.cardkeeper.data.model.ScannedCodeModel
import kotlinx.coroutines.flow.Flow

interface ScannedCodeService : SavedItemService<ScannedCodeModel> {

    fun getScannedCode(codeId: Int): Flow<ScannedCodeEntity>

    fun addScannedCode(scannedCode: ScannedCodeEntity): Flow<ScannedCodeEntity>

    fun updateScannedCode(scannedCode: ScannedCodeEntity): Flow<ScannedCodeEntity>

}